### PR TITLE
fix: various custom operator conformance fixes

### LINKED ISSUE #379: Update flagd-testbed to v3.6.0 and fix in-process evaluation edge cases
## Summary

Update the flagd-testbed dependency to [v3.6.0](https://github.com/open-feature/flagd-testbed/releases/tag/v3.6.0) and fix the in-process evaluation to pass all new edge-case tests.

Testbed v3.6.0 adds coverage for the edge cases identified in:
- open-feature/flagd#1873 (sem_ver parsing inconsistencies)
- open-feature/flagd#1874 (custom operator error return values)
- open-feature/flagd#1875 ($evaluators/$ref resolution gaps)

### sem_ver (open-feature/flagd#1873)

1. **sem_ver build metadata not ignored in comparisons**; uses `==`/`<` rich comparison operators which include build metadata, violating SemVer 2.0.0 Rule 10 (e.g. `"1.0.0+build1" == "1.0.0"` returns `false` instead of `true`); needs custom comparison logic that strips build metadata

### Custom operator error returns (open-feature/flagd#1874)

2. **Mixed `None`/`False` error returns across operators**; `starts_with` and `ends_with` return `None` for some error types and `False` for others; should consistently return `None`

### $evaluators/$ref resolution (open-feature/flagd#1875)

3. **No diagnostic feedback on unresolved `$ref`s at parse time**; failure only surfaces at evaluation time via `ReferenceError`
4. **$ref regex relies on `json.dumps()` output format**; no tolerance for whitespace variations in input

Parent: open-feature/flagd#1904

### PR DESCRIPTION
**Pull request by leakonvalinka**
* support partial versions in sem_ver
* return null instead of false for non-string inputs for starts_with and ends_with
* rename internal `parse_version` to `normalize_version`

⚠️ wait until flagd/RPC is updated to merge and remove the added exclude-feature tags in RPC

Fixes: https://github.com/open-feature/python-sdk-contrib/issues/379

---

Related:
* https://github.com/open-feature/flagd/pull/1950
* https://github.com/open-feature/java-sdk-contrib/pull/1778
* https://github.com/open-feature/js-sdk-contrib/pull/1519
* https://github.com/open-feature/dotnet-sdk-contrib/pull/635
* https://github.com/open-feature/go-sdk-contrib/pull/874

### CODE REVIEW

File: providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/custom_ops.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
     JsonPrimitive,
     ends_with,
     fractional,
-    parse_version,
+    normalize_version,
```
[the remark this answers is not in the document]
Comment (leakonvalinka) [reply providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/custom_ops.py:9]: Not sure about this one, is this necessary? I changed it locally but the pre-commit hook reverted it :D

[the remark this answers is not in the document]
Comment (gruebel) [reply providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/custom_ops.py:9]: i think it is is ok. this is an internal function, I don't think anyone relied on it. just to be a bit more carful let's add it as a change in the description.

File: tools/openfeature-flagd-core/tests/test_targeting.py [hunk 2]
```diff
@@ -113,6 +117,9 @@ def test_minor_no_match(self) -> None:
     def test_v_prefix(self) -> None:
         assert sem_ver({}, "v2.0.0", "=", "2.0.0") is True
 
+    def test_partial_version(self) -> None:
```
Comment (gruebel) [inline tools/openfeature-flagd-core/tests/test_targeting.py]: can you also add the `v2.1` -. `2.1.0` test?

**Review by gruebel — APPROVED** [review]
thanks, looks good 🍻

**Review by aepfli — APPROVED** [review]

File: providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/custom_ops.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (leakonvalinka) [reply providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/custom_ops.py:9]: So you mean that we can keep it as it is right now? Sorry, I'm not entirely sure from your answer.

File: tools/openfeature-flagd-core/tests/test_targeting.py (hunk 2 above)
Comment (leakonvalinka) [reply tools/openfeature-flagd-core/tests/test_targeting.py]: Sure, will do!

**Review by marcozabel — APPROVED** [review]
